src/Hacks/thirdperson.cpp

```diff
@@ -1,71 +1,91 @@
 #include "thirdperson.h"
-
-#include "../settings.h"
-#include "../interfaces.h"
+#include "antiaim.h"
 
 bool Settings::ThirdPerson::enabled = false;
 float Settings::ThirdPerson::distance = 30.f;
+ShowedAngles Settings::ThirdPerson::type = ShowedAngles::REAL;
 
 void ThirdPerson::OverrideView(CViewSetup *pSetup)
 {
-	if (!Settings::ThirdPerson::enabled) {
+	C_BasePlayer* localplayer = (C_BasePlayer*) entityList->GetClientEntity(engine->GetLocalPlayer());
+
+	if(!localplayer)
+		return;
+
+	C_BaseCombatWeapon* activeWeapon = (C_BaseCombatWeapon*) entityList->GetClientEntityFromHandle(localplayer->GetActiveWeapon());
+
+	if (activeWeapon && activeWeapon->GetCSWpnData() && activeWeapon->GetCSWpnData()->GetWeaponType() == CSWeaponType::WEAPONTYPE_GRENADE)
+	{
 		input->m_fCameraInThirdPerson = false;
 		return;
 	}
 
-	C_BasePlayer *localplayer = (C_BasePlayer*) entityList->GetClientEntity(engine->GetLocalPlayer());
-	if(!localplayer || !localplayer->GetAlive())
-		return;
+	if(localplayer->GetAlive() && Settings::ThirdPerson::enabled && !engine->IsTakingScreenshot())
+	{
+		QAngle viewAngles;
+		engine->GetViewAngles(viewAngles);
+		trace_t tr;
+		Ray_t traceRay;
+		Vector eyePos = localplayer->GetEyePosition();
 
-	QAngle *view = localplayer->GetVAngles();
-	trace_t tr;
-	Ray_t ray;
+		Vector camOff = Vector(cos(DEG2RAD(viewAngles.y)) * Settings::ThirdPerson::distance,
+							   sin(DEG2RAD(viewAngles.y)) * Settings::ThirdPerson::distance,
+							   sin(DEG2RAD(-viewAngles.x)) * Settings::ThirdPerson::distance);
 
-	Vector desiredCamOffset = Vector(cos(DEG2RAD(view->y)) * Settings::ThirdPerson::distance,
-								  sin(DEG2RAD(view->y)) * Settings::ThirdPerson::distance,
-								  sin(DEG2RAD(-view->x)) * Settings::ThirdPerson::distance
-							);
+		traceRay.Init(eyePos, (eyePos - camOff));
+		CTraceFilter traceFilter;
+		traceFilter.pSkip = localplayer;
+		trace->TraceRay(traceRay, MASK_SHOT, &traceFilter, &tr);
 
-	//cast a ray from the Current camera Origin to the Desired 3rd person Camera origin
-	ray.Init(localplayer->GetEyePosition(), (localplayer->GetEyePosition() - desiredCamOffset));
-	CTraceFilter traceFilter;
-	traceFilter.pSkip = localplayer;
-	trace->TraceRay(ray, MASK_SHOT, &traceFilter, &tr);
+		Vector eyeDifference = eyePos - tr.endpos;
 
-	Vector diff = localplayer->GetEyePosition() - tr.endpos;
+		float distance2D = sqrt(abs(eyeDifference.x * eyeDifference.x) + abs(eyeDifference.y * eyeDifference.y));
```

Comment:
impossible for squared numbers to be negative, don't need abs()

also this is what the .Length2D() function does